pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -992,37 +755,24 @@ private IdealState setupTable(TableConfig tableConfig, IdealState idealState, in
    * We check the segment's metadata to see if that is old enough for repair. If it is fairly new, we
    * leave it as it is, to be fixed the next time repair job triggers.
    */
-  @VisibleForTesting
-  protected boolean isTooSoonToCorrect(String tableNameWithType, String segmentId, long now) {
-    Stat stat = new Stat();
-    LLCRealtimeSegmentZKMetadata metadata = getRealtimeSegmentZKMetadata(tableNameWithType, segmentId, stat);
-    long metadataUpdateTime = stat.getMtime();
-    if (now < metadataUpdateTime + MAX_SEGMENT_COMPLETION_TIME_MILLIS) {
-      LOGGER.info("Too soon to correct segment:{} updateTime: {} now:{}", segmentId, metadataUpdateTime, now);
-      return true;
-    }
-    return false;
-  }
 
   /**
-   *
    * Returns true if more than {@link PinotLLCRealtimeSegmentManager::MAX_SEGMENT_COMPLETION_TIME_MILLIS} ms have elapsed since segment metadata update
    */
-  private boolean isExceededMaxSegmentCompletionTime(String tableNameWithType, String segmentId, long now) {
+  @VisibleForTesting
+  boolean isExceededMaxSegmentCompletionTime(String realtimeTableName, String segmentName, long currentTimeMs) {
     Stat stat = new Stat();
-    LLCRealtimeSegmentZKMetadata metadata = getRealtimeSegmentZKMetadata(tableNameWithType, segmentId, stat);
-    long metadataUpdateTime = stat.getMtime();
-    if (now > metadataUpdateTime + MAX_SEGMENT_COMPLETION_TIME_MILLIS) {
-      LOGGER
-          .info("Segment:{}, Now:{}, metadataUpdateTime:{}, Exceeded MAX_SEGMENT_COMPLETION_TIME_MILLIS:{}", segmentId,
-              now, metadataUpdateTime, MAX_SEGMENT_COMPLETION_TIME_MILLIS);
-      return true;
-    }
-    return false;
+    getSegmentZKMetadata(realtimeTableName, segmentName, stat);
+    return currentTimeMs > stat.getMtime() + MAX_SEGMENT_COMPLETION_TIME_MILLIS;
```

Comment:
We found it useful to log here. Can you please add back the log?

Reply:
Added back